api/course.js

```diff
@@ -4,19 +4,63 @@
 require("path");
 const express = require("express");
 const app = express();
-const {getCourse} = require("../models/course");
+const {getCourse, getLiveCourses} = require("../models/course");
+const {getUserById} = require("../models/user");
+const {Role} = require("../entities/role");
 const {requireAuth} = require("../services/auth/auth");
 
+// Get all courses from the Course API server,
+// then process the collected courses so that they conform with our database.
+// Then add each course to the database that doesn't already exist.
+// If the course already exists we update it instead.
+app.get("/updateDatabase", requireAuth, async (req, res) => {
+
+  try {
+
+    console.log("Get all courses from the OSU course API");
+
+    // get user data
+    const authenticatedUser = await getUserById(req.auth.userId);
+
+    // due to the computational cost of this action,
+    // only the head advisor can perform this action
+    if (authenticatedUser.role === Role.advisor ||
```

Comment:
Should advisors be able to perform admin operations like updating courses?

Reply:
Good catch, you are correct that only the head advisor should be able to perform this action.